### PR TITLE
Manually expire cookies after a configured amount of time

### PR DESCRIPTION
**Pull request by parijke**
Add listeners to replace the "old" handlers to manage cookie expiration. The values in     `session_max_absolute_lifetime:` and `session_max_relative_lifetime:` will be used to calculate expiration

### CODE REVIEW

File: bin/console [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env php
 <?php
 
-use Surfnet\StepupRa\Kernel;
+use src\Kernel;
```
Comment (MKodde) [inline bin/console:4]: The other projects solved this by moving the kernel to the Surfnet\StepupRa folder. I much prefer this to introducing a bogus /src namespace. Can u fix?

File: config/services.yaml [hunk 2]
```diff
@@ -38,3 +38,13 @@ services:
     arguments:
       - '@security.helper'
       - "%logout_redirect_url%"
+
+  Surfnet\StepupRa\RaBundle\EventListener\AuthenticatedUserListener:
+
+  Surfnet\StepupRa\RaBundle\EventListener\ExplicitSessionTimeoutListener:
```
Comment (MKodde) [inline config/services.yaml:44]: Are these entries needed? Autoconfigure/autowire should pick this up without issue

File: src/Surfnet/StepupRa/RaBundle/EventListener/AuthenticatedUserListener.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        private readonly AuthenticatedSessionStateHandler $sessionStateHandler,
+        private readonly LoggerInterface                  $logger,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            // The firewall, which makes the token available, listens at P8
+            // We must jump in after the firewall, forcing us to overwrite the translator locale.
```
Comment (MKodde) [inline src/Surfnet/StepupRa/RaBundle/EventListener/AuthenticatedUserListener.php:46]: Again, the `translator locale` part of this comment seems like a copy paste from another class?

File: src/Surfnet/StepupRa/RaBundle/EventListener/AuthenticatedUserListener.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Copyright 2016 SURFnet bv
```
Comment (MKodde) [inline src/Surfnet/StepupRa/RaBundle/EventListener/AuthenticatedUserListener.php:6]: Nitpick ~2016~ 2024

File: src/Surfnet/StepupRa/RaBundle/EventListener/ExplicitSessionTimeoutListener.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        private LoggerInterface                  $logger,
+        private EventDispatcherInterface         $eventDispatcher,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            // The firewall, which makes the token available, listens at P8
+            // We must jump in after the firewall, forcing us to overwrite the translator locale.
```
Comment (MKodde) [inline src/Surfnet/StepupRa/RaBundle/EventListener/ExplicitSessionTimeoutListener.php:53]: See previous comment

**Review by MKodde — CHANGES_REQUESTED** [review]
Thanks for porting this to Stepup RA! See some feedback below

File: config/services.yaml (hunk 2 above)
Comment (parijke) [reply config/services.yaml:44]: Yes, without it won't work as the  Surfnet\StepupRa\RaBundle\* is not in the search path

File: src/Surfnet/StepupRa/RaBundle/EventListener/AuthenticatedUserListener.php (hunk 3 above)
Comment (parijke) [reply src/Surfnet/StepupRa/RaBundle/EventListener/AuthenticatedUserListener.php:46]: removed the copied comments

File: src/Surfnet/StepupRa/RaBundle/EventListener/ExplicitSessionTimeoutListener.php (hunk 5 above)
Comment (parijke) [reply src/Surfnet/StepupRa/RaBundle/EventListener/ExplicitSessionTimeoutListener.php:53]: removed comments

File: bin/console (hunk 1 above)
Comment (MKodde) [reply bin/console]: Thanks that looks much better!

**Review by MKodde — APPROVED** [review]
Lookin' good. Please, write a small PR description before merging and tagging! I already set a more descriptive PR title, feel free to tweak that too.

**Pull request merged**